menelaus/injection/class_manipulation.py

```diff
@@ -94,3 +96,119 @@ def class_join(data, target_col, class_1, class_2, new_class, from_index, to_ind
         class_idx = class_idx[(class_idx < to_index) & (class_idx >= from_index)]
         ret[class_idx, target_col] = new_class
     return ret
+
+# endregion
+
+
+# region - LTF-inspired class manipulation functions
+
+def class_probability_shift(data, target_col, from_index, to_index, classes, p_classes):
+    """
+    Resamples the data over a specified window, with altered probability
+    for specified classes (and uniform probability for remaining classes).
+    Accepts ``pandas.DataFrame`` with column names or ``numpy.ndarray``
+    with column indices.
+
+    Ref. :cite:t:`LTFmethods`
+
+    Args:
+        data (np.array): data to inject with drift
+        target_col (int or str): column index/label of targets column
+        from_index: row index at which to start class swap
+        to_index: row index at which to end (non-inclusive) class swap
+        classes (list[int]): list of classes with altered sampling chance
+        p_classes (float): altered sampling chance for members of ``classes``
+
+    Returns:
+        np.array or pd.DataFrame: copy of data, resampled with shifted
+            class probability for 1 or more desired classes
+    """
+    ret = np.copy(data)
+    classes = np.unique(ret[:, target_col])
+
+    # distribution for each data point, and reordering of each point by class
+    p_distribution = np.array([])
+    sample_grouped = np.array([])
+
+    # locate all classes to shift
+    shifted_classes_idx = np.where(data[:, target_col].isin(classes))[0]
```

Comment:
I tried testing out these functions with a dataset on my end to get a better understanding of how they work. I’m encountering a few errors related to the data type. The input for this one is np.array input. I got an error as I think 'isin' only works with pd.dataframes? 

Do you think we should set up this function work with both pd.dataframes and np.arrays? Or just one of them. If you have a busy week and want help adjusting the syntax, let me know.

Reply:
I think they definitely should support both, as the original functions I wrote do. 

It looks like I haven't quite tested even the `numpy` logic yet, but for these new functions I was thinking of just doing a DF => `numpy` => back to DF conversion to handle the `pandas` case.